st2common/st2common/util/action_db.py

```diff
@@ -213,14 +218,14 @@ def get_args(action_parameters, action_db):
     positional_args = []
     positional_args_keys = set()
     for _, arg in six.iteritems(position_args_dict):
-        positional_args.append(str(action_parameters.get(arg)))
+        positional_args.append(_quote_text_lines(str(action_parameters.get(arg))))
```

Comment:
You may want to use quote_unix from st2common.util.shell. 

Example:

In [3]: from st2common.util.shell import quote_unix

In [4]: quote_unix('foo\nbar whatevs')
Out[4]: "'foo\nbar whatevs'"

In [5]:

Also, to make sure nothing is broken, run ssh tests from st2tests. 

https://github.com/StackStorm/st2tests/blob/master/packs/tests/actions/test_quickstart_remote_script_actions.yaml

https://github.com/StackStorm/st2tests/blob/master/packs/tests/actions/test_quickstart_local_script_actions.yaml